### ISSUE TITLE
Bad slicing for some planes

### ISSUE THREAD

**Issue by julienr** [julienr/meshcut#7]
Follow-up from https://github.com/vlachoudis/bCNC/issues/901

In `examples/1_stl_sphere_cut.py`, with the settings

```
plane_orig = (0, 0.75, 0)
plane_norm = (0, 1, 0)
```

this gives :

<img width="512" alt="screen shot 2018-09-08 at 10 26 43" src="https://user-images.githubusercontent.com/506602/45252095-b11d2400-b351-11e8-8899-5b927534ba32.png">

**Comment by Harvie** [comment]
Any ideas what might cause this?

**Comment by julienr** [comment]
Trying to reproduce this: is this still with the `hemisphere.stl` model ? The plane `y=14.5` doesn't intersect the sphere, so I guess there is some scaling going on.

**Comment by Harvie** [comment]
I am using this model:

https://github.com/vlachoudis/bCNC/files/2358797/sphere.stl.zip

But i've got very similar problems with sphere.stl from your repository. see following:
https://github.com/vlachoudis/bCNC/issues/901#issuecomment-419243166

**Comment by Harvie** [comment]
@julienr note it is Y -14.5 not Y 14.5

**Comment by Harvie** [comment]
I've tried increasing `close_epsilon` and `dist_tol` magic numbers and it does not seem to help.

**Comment by julienr** [comment]
I am not able to reproduce this on the `1_stl_sphere_cut.py` example of meshcut. Here is an example with the `sphere.stl.zip` that you gave above and

```
    plane_orig = (0, -14.5, 0)
    plane_norm = (0, 1, 0)
```
<img width="750" alt="screen shot 2018-10-24 at 21 57 38" src="https://user-images.githubusercontent.com/506602/47457613-da95f000-d7d7-11e8-8c0b-5e0c03a0431f.png">

I tried to play a bit with various parameters, but couldn't find some that would reproduce the bug. Are you sure this is not a display problem in bCNC ?

**Comment by Harvie** [comment]
Update: i had problem in bCNC. I had typo in code which automaticaly closes the contour. Now i fixed it:

![image](https://user-images.githubusercontent.com/276504/47610378-ef7eb780-da53-11e8-9a7b-904fec983cb8.png)

I think that meshcut should close the path itself if needed. Now i close it by returning to original coordinates, but this might not be desirable for non-watertight meshes.

```python
first = contour[0]
block.append("g0 x%f y%f z%f"%(first[0],first[1],first[2]))
for segment in contour:
	block.append("g1 x%f y%f z%f"%(segment[0],segment[1],segment[2]))
block.append("g1 x%f y%f z%f"%(first[0],first[1],first[2])) #HERE I CLOSE THE PATH
```


Anyway... i still have problem in y = 0:

![image](https://user-images.githubusercontent.com/276504/47610358-b6dede00-da53-11e8-9abb-96d0440f53c9.png)

**Comment by Harvie** [comment]
This last problem seems to be fixed by updating to latest version of meshcut. I am quite happy with how it turned out. Only issue is the non-manifold case. It's not really critical, as most meshes are manifold, however it would make meshcut more consistent if fixed.

**Comment by unre4l** [comment]
any news regarding this issue? 🤔

**Comment by julienr** [comment]
@unre4l Are you running into this with a non-manifold mesh ? Do you have a repro mesh ?

I've been quite busy, but I think fixing this would involve keeping track of the starting triangle and the ending triangle and if those are the same triangle, then it means the mesh is manifold and we can close the path. If the two triangles are different, then the mesh is non-manifold and the path shouldn't be closed. If you want to work on it, PR welcome. Otherwise, I can have a look, but I don't promise anything :)

**Comment by julienr** [comment]
Duplicate of #20, closed by #21
